Sources/Interaction/Widgets/AbstractWidget/index.js

```diff
@@ -11,105 +8,41 @@ const { vtkErrorMacro } = macro;
 
-  publicAPI.listenEvents = () => {
-    if (!model.interactor) {
-      vtkErrorMacro('The interactor must be set before listening events');
-      return;
+  publicAPI.setInteractor = (i) => {
+    // If we already have an interactor then remove representation
+    if (model.interactor && model.widgetRep) {
+      model.currentRenderer.removeViewProp(model.widgetRep);
     }
 
-    // Remove current events
-    while (model.unsubscribes.length) {
-      model.unsubscribes.pop().unsubscribe();
-    }
+    // Enable/disable in superclass
+    superClass.setInteractor(i);
 
-    // Check what events we can handle and register callbacks
-    vtkRenderWindowInteractor.handledEvents.forEach((eventName) => {
-      if (publicAPI[`handle${eventName}`]) {
-        model.unsubscribes.push(
-          model.interactor[`on${eventName}`](
-            publicAPI[`handle${eventName}`],
-            model.priority
-          )
-        );
-      }
-    });
-  };
-
-  publicAPI.setInteractor = (i) => {
-    if (i === model.interactor) {
+    // Add representation to new interactor's renderer
+    if (!model.currentRenderer) {
       return;
     }
-    model.interactor = i;
-
-    if (i && model.enabled) {
-      publicAPI.listenEvents();
-    }
-
-    publicAPI.modified();
+    publicAPI.createDefaultRepresentation();
+    model.widgetRep.setRenderer(model.currentRenderer);
+    model.widgetRep.buildRepresentation();
+    model.currentRenderer.addViewProp(model.widgetRep);
   };
 
+  //----------------------------------------------------------------------------
   publicAPI.render = () => {
     if (!model.parent && model.interactor) {
       model.interactor.render();
     }
   };
 
-  publicAPI.setEnable = (enable) => {
-    if (enable === model.enabled) {
-      return;
-    }
-
-    if (enable) {
-      if (!model.interactor) {
-        vtkErrorMacro(
-          'The interactor must be set prior to enabling the widget'
-        );
-        return;
-      }
-
-      const eventPosition = model.interactor.getEventPosition();
-      let X = 0;
-      let Y = 0;
-      if (eventPosition) {
-        X = eventPosition[0];
-        Y = eventPosition[1];
-      }
-
-      if (!model.currentRenderer) {
-        model.currentRenderer = model.interactor.findPokedRenderer(X, Y);
-        if (!model.currentRenderer) {
-          return;
-        }
-      }
-      model.enabled = 1;
-      publicAPI.createDefaultRepresentation();
-      model.widgetRep.setRenderer(model.currentRenderer);
-
-      // Enable listening events
-      publicAPI.listenEvents();
-
-      model.widgetRep.getBounds();
-      model.widgetRep.buildRepresentation();
-      model.currentRenderer.addViewProp(model.widgetRep);
-    } else {
-      model.enabled = 0;
-
-      // Don't listen to events
-      while (model.unsubscribes.length) {
-        model.unsubscribes.pop().unsubscribe();
-      }
-
-      if (model.currentRenderer) {
-        model.currentRenderer.removeViewProp(model.widgetRep);
```

Comment:
Should we still hide it when disabling?

Reply:
don't know what does VTK do?

Reply:
AFAIK VTK removes the prop like above.

Interestingly, LineWidget uses setEnabled(false) to turn off the widget, and then re-adds the representation into the scene. This has the effect of disabling widget interaction while keeping the widget in-scene. Maybe we can add a method like `setInteractionEnabled` so widgets can disable interactions, but keep the representation in the renderer?

Reply:
For the next round of things :)